packages/apollo-collaboration-server/src/features/features.service.ts

```diff
@@ -362,4 +363,49 @@ export class FeaturesService {
     }
     return featureIdArrAsParam
   }
+
+  /**
+   * Fetch features based on Reference seq, Start and End -values
+   * @param request - Contain search criteria i.e. refname, start and end -parameters
+   * @returns Return 'HttpStatus.OK' and array of features if search was successful
+   * or if search data was not found or in case of error throw exception
+   */
+  async getFeaturesByCriteria(searchDto: FeatureRangeSearchDto) {
+    // Search correct refSeqs by assemblyId
+    const refSeqs = await this.refSeqModel
+      .find({ assemblyId: searchDto.assemblyId })
+      .exec()
+
+    if (!refSeqs) {
+      const errMsg = `ERROR: No RefSeqs were found for assemblyId: ${searchDto.assemblyId}`
+      this.logger.error(errMsg)
+      throw new NotFoundException(errMsg)
+    }
+    const refSeqIdIdArray = refSeqs.map((refSeq) => refSeq._id)
+    this.logger.debug(`Found refSeqs: ${refSeqIdIdArray}`)
+
+    // Search correct feature
+    const features = await this.featureModel
+      .find({
+        'gff3FeatureLineWithRefs.seq_id': searchDto.refName,
+        'gff3FeatureLineWithRefs.start': { $lte: searchDto.end },
+        'gff3FeatureLineWithRefs.end': { $gte: searchDto.start },
+        seq_id: refSeqIdIdArray,
+      })
+      .exec()
+    this.logger.debug(
+      `Searching features for AssemblyId: ${searchDto.assemblyId}, refName: ${searchDto.refName}, start: ${searchDto.start}, end: ${searchDto.end}`,
+    )
+
+    if (!features) {
```

Comment:
Will this ever be undefined? If there are no matches, it should be an empty array, right?

Reply:
As background:
**_The findOne() returns first document if query matches otherwise returns null. The find() method does not return null, it returns a cursor_**
So, after we make query we have to check if return value was NULL or in case of find() -method, we have to check the length of cursor (i.e. is it less than one = no docs found)